torch/masked/_ops.py

```diff
@@ -1385,12 +1385,7 @@ def mean(
             total = sum(input, dim, keepdim=keepdim, dtype=dtype)
         else:
             inmask = _input_mask(input, mask=mask)
-            count = sum(
-                inmask.new_ones(input.shape, dtype=torch.int64),
-                dim,
-                keepdim=keepdim,
-                mask=inmask,
-            )
+            count = inmask.sum(dim=dim, keepdim=keepdim)
```

Comment:
If `inmask` is bool, doing sum currently would also upcast/allocate a not very required temp copy I think:
- https://github.com/pytorch/pytorch/issues/55366

Reply:
I think it is guaranteed to be bool. Sorry if I'm not understanding, but did you have a suggested change to make?

Reply:
I'm not a maintained, so my personal opinion is: it's already nice that the PR fixes the scaling problem, but maybe this issue that `bool_tensor.sum()` allocates a temp/upcast copy of the input in eager mode could be bumped